src/links.ts

```diff
@@ -63,7 +63,23 @@ export function getLinks(source: string, baseUrl: string): ParsedUrl[] {
   const sanitized = links
     .filter(link => !!link)
     .map(link => parseLink(link, realBaseUrl));
-  return sanitized;
+  // Filter out the links with the same url.href like
+  // /page /page#fragment1 /page#fragment2
+  const uniqueHref = new Set();
+  const uniqueLinks: ParsedUrl[] = [];
+  sanitized.forEach(link => {
+    // ignore the links without a url
+    if (!link.url) {
```

Comment:
Is this a thing that happens? The filter up above `.filter(link => !!link)` should have had the same effect, I would think

Reply:
in `index.ts` it has this: 
```
      const urlResults = getLinks(data, opts.url.href);
      for (const result of urlResults) {
        // if there was some sort of problem parsing the link while
        // creating a new URL obj, treat it as a broken link.
        if (!result.url) {
        ....
        }
```
so I thought it might be a case that link can be without URL? But I agree `!!link` will check it first and filter out that situation.